### PR TITLE
Fix issues seen during delta-rs integration

### PR DESCRIPTION
**Pull request by yjshen**

### CODE REVIEW

File: src/hdfs.rs [hunk 1]
```diff
@@ -647,23 +647,27 @@ impl HdfsFile {
 
     /// Write data into an open file.
     pub fn write(&self, buf: &[u8]) -> Result<i32, HdfsErr> {
```
Comment (yjshen) [inline src/hdfs.rs:649]: To align with the semantics of LocalFileSystem write, and imitate the behavior of https://doc.rust-lang.org/src/std/io/mod.rs.html#1540-1555, which actually allows an empty buffer as input, but writes the whole buffer out.

File: src/hdfs.rs [hunk 2]
```diff
@@ -222,7 +222,7 @@ impl HdfsFs {
         };
 
         if ptr.is_null() {
-            Err(HdfsErr::Generic(format!(
+            Err(HdfsErr::FileNotFound(format!(
```
Comment (yjshen) [inline src/hdfs.rs:225]: Mistyped error.

File: src/hdfs.rs [hunk 3]
```diff
@@ -173,7 +173,7 @@ impl HdfsFs {
     /// Create HdfsFile from hdfsFile
     fn new_hdfs_file(&self, path: &str, file: hdfsFile) -> Result<HdfsFile, HdfsErr> {
         if file.is_null() {
-            Err(HdfsErr::Generic(format!(
+            Err(HdfsErr::FileNotFound(format!(
```
Comment (yjshen) [inline src/hdfs.rs:176]: Mistyped error.

File: src/hdfs.rs (hunk 1 above)
Comment (yahoNanJing) [reply src/hdfs.rs:649]: As https://github.com/datafusion-contrib/fs-hdfs/blob/cadaee54e20f9c09484dae088485908ece966893/c_src/libhdfs/hdfs.c#L1464-L1466 mentioned, the `hdfsWrite` won't write partial contents. Maybe it's better to refine the condition check as follows:
`
if written_len >= 0 {
            Ok(written_len)
        } else {
            Err(HdfsErr::Generic(format!(
                "Fail to write contents to file {}",
                self.path()
            )))
        }
`

Comment (yjshen) [reply src/hdfs.rs:649]: Thanks, will do

Comment (yjshen) [reply src/hdfs.rs:649]: Reverted, only allow to call with an empty buffer and do nothing for this case.

**Review by yahoNanJing — APPROVED** [review]
LGTM

**Pull request merged**